pdns/ws-auth.cc

```diff
@@ -737,7 +740,7 @@ static void updateDomainSettingsFromDocument(UeberBackend& B, const DomainInfo&
         if (!dk.unSecureZone(zonename, error, info)) {
           throw ApiException("Error while un-securing zone '"+ zonename.toString()+"': " + error);
         }
-        isDNSSECZone = dk.isSecuredZone(zonename);
+        isDNSSECZone = dk.isSecuredZone(zonename, false); 
```

Comment:
there's a similar call in the other branch (line 726), that could use it as well?

Reply:
The short answer is no. The longer version is...
There was no metadata update inside the transaction at line 726, so the cache is valid and it is also fine to fill the cache when expired.
Here the unsetNSEC3PARAM(); and unsetPresigned(); in unSecureZone() are responsible for a write operation inside the transaction, so the isPresigned() in side isSecuredZone() is no longer allowed to read/update the cache.